### PR TITLE
NAS-108535 / 12.0 / Nas 108535

### PR DESCRIPTION
**Pull request by ryousaf**

### CODE REVIEW

File: src/app/pages/storage/disks/disk-list/disk-list.component.ts [hunk 1]
```diff
@@ -120,9 +121,11 @@ export class DiskListComponent {
 		this.ws.call('disk.get_unused', []).subscribe((unused_res) => {
 			this.unused = unused_res;
 		});
+		this.ws.call('smart.test.disk_choices').subscribe(res => this.SMARTdiskChoices = res)
```
Comment (william-gr) [inline src/app/pages/storage/disks/disk-list/disk-list.component.ts:124]: Does something logs if this call fails?

Comment (ryousaf) [reply src/app/pages/storage/disks/disk-list/disk-list.component.ts]: Added error handling using EntityUtils

File: src/app/pages/task-calendar/smart/smart-form/smart-form.component.ts [hunk 2]
```diff
@@ -102,16 +102,19 @@ export class SmartFormComponent {
...
+              protected ws: WebSocketService,
               protected taskService: TaskService,
               protected storageService: StorageService,
               protected entityFormService: EntityFormService,
               protected aroute: ActivatedRoute) {
     this.disk_field = _.find(this.fieldSets[0].config, { 'name': 'disks' });
-    this.storageService.listDisks().subscribe((res) => {
-      for (const i in res) {
-        this.disk_field.options.push({ label: res[i].name, value: res[i].identifier })
+    this.ws.call('smart.test.disk_choices').subscribe(
```
Comment (william-gr) [inline src/app/pages/task-calendar/smart/smart-form/smart-form.component.ts:111]: I believe we also need error handling here?

Comment (ryousaf) [reply src/app/pages/task-calendar/smart/smart-form/smart-form.component.ts:112]: Yes. Fixed this as well.

**Review by dszidi — APPROVED** [review]
LGTM

**Pull request merged**